### PR TITLE
Create ServerLRAFilter as a CDI bean.

### PR DESCRIPTION
**Pull request by idlewis**
It should have been like this from the start

### CODE REVIEW

**Comment by idlewis** [comment]
This is currently blocked by #14883

File: dev/io.openliberty.microprofile.lra.1.0.internal/bnd.bnd [hunk 1]
```diff
@@ -45,6 +45,9 @@ Include-Resource: \
 	com.ibm.ws.org.osgi.annotation.versioning;version=latest, \
 	com.ibm.wsspi.org.osgi.service.component.annotations;version=latest, \
 	com.ibm.websphere.javaee.jaxrs.2.1;version=latest, \
-	com.ibm.ws.jaxrs.2.1.common;version=latest
+	com.ibm.ws.jaxrs.2.1.common;version=latest, \
+	com.ibm.websphere.javaee.cdi.2.0
+	#com.ibm.ws.cdi.interfaces;version=latest, \
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.lra.1.0.internal/bnd.bnd:50]: delete if not needed

File: dev/io.openliberty.microprofile.lra.1.0.internal/src/io/openliberty/microprofile/lra/internal/LraFilterComponent.java [hunk 2]
```diff
@@ -99,14 +101,8 @@ public void installProvider(boolean clientSide, List<Object> providers, Set<Stri
...
-            try {
-                // Rather unhelpfully, the ServerLRAFilter constructor throws 'Exception'. There isn't much we can do with
-                // that. Re-throwing should prevent the servlet being initialized, which is probably better than swallowing
-                // the exception
-                providers.add(new ServerLRAFilter());
-            } catch (Exception e) {
-                throw new LraRuntimeException(Tr.formatMessage(tc, "LRA_CANT_REGISTER_FILTERS.CWMRX5001E", e), e);
-            }
+            ServerLRAFilter beanInstance = CDI.current().select(ServerLRAFilter.class).get();
+            providers.add(beanInstance);
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.lra.1.0.internal/src/io/openliberty/microprofile/lra/internal/LraFilterComponent.java:105]: Don't you still need some error handling if the bean can not be found?

File: dev/io.openliberty.org.jboss.narayana.rts/src/io/openliberty/microprofile/lra/internal/CdiExtensionComponent.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package io.openliberty.microprofile.lra.internal;
+/*******************************************************************************
```
Comment (tevans78) [inline dev/io.openliberty.org.jboss.narayana.rts/src/io/openliberty/microprofile/lra/internal/CdiExtensionComponent.java:2]: better to put the copyright above the package statement

File: dev/io.openliberty.org.jboss.narayana.rts/bnd.bnd [hunk 4]
```diff
@@ -22,8 +22,21 @@ Export-Package: \
```
Comment (tevans78) [inline dev/io.openliberty.org.jboss.narayana.rts/bnd.bnd:22]: I can't see any bundles which import these packages?? Are these exports actually needed, should the packages be private instead?

Comment (idlewis) [reply dev/io.openliberty.org.jboss.narayana.rts/bnd.bnd:22]: They are imported implicitly by the buildpath statement in io.openliberty.microprofile.lra.1.0.internal

File: dev/io.openliberty.microprofile.lra.1.0.internal/bnd.bnd (hunk 1 above)
Comment (idlewis) [reply dev/io.openliberty.microprofile.lra.1.0.internal/bnd.bnd:50]: will remove

File: dev/io.openliberty.org.jboss.narayana.rts/src/io/openliberty/microprofile/lra/internal/CdiExtensionComponent.java (hunk 3 above)
Comment (idlewis) [reply dev/io.openliberty.org.jboss.narayana.rts/src/io/openliberty/microprofile/lra/internal/CdiExtensionComponent.java:2]: moved

**Review by tevans78 — APPROVED** [review]

**Comment by idlewis** [comment]
Squashed review comments and repushed, but no additional changes made

**Comment by idlewis** [comment]
#build

**Comment by idlewis** [comment]
#libbybot

**Pull request merged**